adapters/base/src/main/java/org/apache/cassandra/sidecar/adapters/base/TokenRangeReplicaProvider.java

```diff
@@ -115,22 +122,45 @@ private List<TokenRangeReplicas> transformRangeMappings(Map<List<String>, List<S
                               .collect(toList());
     }
 
-    private Map<String, String> replicaToStateMap(List<TokenRangeReplicas> replicaSet, StorageJmxOperations storage)
+    private List<ReplicaMetadata> getReplicaMetadata(List<TokenRangeReplicas> replicaSet,
+                                                     StorageJmxOperations storage,
+                                                     Map<String, String> hostToDatacenter)
     {
         List<String> joiningNodes = storage.getJoiningNodesWithPort();
         List<String> leavingNodes = storage.getLeavingNodesWithPort();
         List<String> movingNodes = storage.getMovingNodesWithPort();
 
+        List<String> liveNodes = storage.getLiveNodesWithPort();
+        List<String> deadNodes = storage.getUnreachableNodesWithPort();
+
+
         String rawGossipInfo = getRawGossipInfo();
         GossipInfoResponse gossipInfo = GossipInfoParser.parse(rawGossipInfo);
 
         StateWithReplacement state = new StateWithReplacement(joiningNodes, leavingNodes, movingNodes, gossipInfo);
+        RingProvider.Status status = new RingProvider.Status(liveNodes, deadNodes);
 
         return replicaSet.stream()
                          .map(TokenRangeReplicas::replicaSet)
                          .flatMap(Collection::stream)
                          .distinct()
-                         .collect(Collectors.toMap(Function.identity(), state::of));
+                         .map(replica -> {
+                             try
+                             {
+                                 HostAndPort hap = HostAndPort.fromString(replica);
+                                 return new ReplicaMetadata(state.of(replica),
+                                                            status.of(replica),
+                                                            dnsResolver.reverseResolve(hap.getHost()),
+                                                            replica,
```

Comment:
since we already have it split, let's add an address field and a port field. 
```suggestion
                                                            hap.getHost(),
                                                            hap.getPort(),
```